src/BatteryStateOfChargeService.cpp

```diff
@@ -43,21 +49,39 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmpUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (newCurrent_ <= 0)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
-}
+    double newTimeSecs = QTime(0,0,0).secsTo(newTime_);
+    double TimeElapsed =(BATTERY_AMP_HOUR_CAPACITY / fabs(newCurrent_)) * SECONDS_TO_HOURS;
+    newTimeSecs = newTimeSecs + TimeElapsed;
+    QTime estimatedTime = QTime(0,0,0).addSecs(newTimeSecs);
 
+    return estimatedTime;
+}
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+    newTime_ = batteryData.time;
+    newCurrent_ = batteryData.current;
+
+    double TimeUsed = newTime_.msecsTo(oldTime_);
+    double averageCurrent = (newCurrent_ + oldCurrent_) / 2;
+    totalAmpUsed_ = initialAmpUsed_ + averageCurrent  / (TimeUsed / SECONDS_TO_MILISECONDS);
```

Comment:
You should be updating `totalAmpUsed_` based on `totalAmpUsed_`. By updating it according to `initialAmpUsed_` you will be having a number similar to the initial amps rather than an updating total amps.